File: csharp/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed <= 0) return 0.0;
        if (speed <= 4) return 1.0;
        if (speed <= 8) return 0.9;
        if (speed <= 9) return 0.8;

        return 0.77;
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        double success = SuccessRate(speed);

        return 221 * speed * success;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(ProductionRatePerHour(speed) / 60);
    }
}

File: fsharp/CarsAssemble.fs
module CarsAssemble

let defaultSpeed = 221

let successRate (speed: int) : float =
    if speed = 0 then 0.0
    elif speed <= 4 then 1.0
    elif speed <= 8 then 0.9
    elif speed = 9 then 0.8
    else 0.77

let productionRatePerHour (speed: int) : float =
    (speed * defaultSpeed |> float) * successRate speed

let workingItemsPerMinute (speed: int) : int =
    (productionRatePerHour speed |> int) / 60
